src/ObjLoading/Game/T6/BSP/Linker/GfxWorldLinker.cpp

```diff
@@ -0,0 +1,772 @@
+        gfxWorld->dpvsDyn.dynEntClientCount[0] = dynEntCount + 256; // the game allocs 256 empty dynents, as they may be used ingame
+        gfxWorld->dpvsDyn.dynEntClientCount[1] = 0;
+
+        // +100: there is a crash that happens when regdolls are created, and dynEntClientWordCount[0] is the issue.
+        // Making the value much larger than required fixes it, but unsure what the root cause is
+        gfxWorld->dpvsDyn.dynEntClientWordCount[0] = ((gfxWorld->dpvsDyn.dynEntClientCount[0] + 31) >> 5) + 100;
+        gfxWorld->dpvsDyn.dynEntClientWordCount[1] = 0;
+        gfxWorld->dpvsDyn.usageCount = 0;
+
+        int dynEntCellBitsSize = gfxWorld->dpvsDyn.dynEntClientWordCount[0] * gfxWorld->dpvsPlanes.cellCount;
+        gfxWorld->dpvsDyn.dynEntCellBits[0] = m_memory.Alloc<unsigned int>(dynEntCellBitsSize);
+        gfxWorld->dpvsDyn.dynEntCellBits[1] = nullptr;
+
+        int dynEntVisData0Size = gfxWorld->dpvsDyn.dynEntClientWordCount[0] * 32;
+        gfxWorld->dpvsDyn.dynEntVisData[0][0] = m_memory.Alloc<char>(dynEntVisData0Size);
+        gfxWorld->dpvsDyn.dynEntVisData[0][1] = m_memory.Alloc<char>(dynEntVisData0Size);
+        gfxWorld->dpvsDyn.dynEntVisData[0][2] = m_memory.Alloc<char>(dynEntVisData0Size);
+        gfxWorld->dpvsDyn.dynEntVisData[1][0] = nullptr;
+        gfxWorld->dpvsDyn.dynEntVisData[1][1] = nullptr;
+        gfxWorld->dpvsDyn.dynEntVisData[1][2] = nullptr;
+
+        unsigned int dynEntShadowVisCount = gfxWorld->dpvsDyn.dynEntClientCount[0] * (gfxWorld->primaryLightCount - gfxWorld->sunPrimaryLightIndex - 1);
+        gfxWorld->primaryLightDynEntShadowVis[0] = m_memory.Alloc<unsigned int>(dynEntShadowVisCount);
+        gfxWorld->primaryLightDynEntShadowVis[1] = nullptr;
+
+        gfxWorld->sceneDynModel = m_memory.Alloc<GfxSceneDynModel>(gfxWorld->dpvsDyn.dynEntClientCount[0]);
+        gfxWorld->sceneDynBrush = nullptr;
+    }
+
+    bool GfxWorldLinker::loadOutdoors(GfxWorld* gfxWorld)
+    {
+        float xRecip = 1.0f / (gfxWorld->maxs.x - gfxWorld->mins.x);
+        float xScale = -(xRecip * gfxWorld->mins.x);
+
+        float yRecip = 1.0f / (gfxWorld->maxs.y - gfxWorld->mins.y);
+        float yScale = -(yRecip * gfxWorld->mins.y);
+
+        float zRecip = 1.0f / (gfxWorld->maxs.z - gfxWorld->mins.z);
+        float zScale = -(zRecip * gfxWorld->mins.z);
+
+        memset(gfxWorld->outdoorLookupMatrix, 0, sizeof(gfxWorld->outdoorLookupMatrix));
+
+        gfxWorld->outdoorLookupMatrix[0].x = xRecip;
+        gfxWorld->outdoorLookupMatrix[1].y = yRecip;
+        gfxWorld->outdoorLookupMatrix[2].z = zRecip;
+        gfxWorld->outdoorLookupMatrix[3].x = xScale;
+        gfxWorld->outdoorLookupMatrix[3].y = yScale;
+        gfxWorld->outdoorLookupMatrix[3].z = zScale;
+        gfxWorld->outdoorLookupMatrix[3].w = 1.0f;
+
+        std::string outdoorImageName = std::string("$outdoor");
+        auto outdoorImageAsset = m_context.LoadDependency<AssetImage>(outdoorImageName);
```

Comment:
nit: since `LoadDependency` takes in a `const std::string&`, you can pass in a literal value directly instead of having to make a temporary variable that you only use once.

adjusted in: 049fa2b8ae999aabfe5f91355ce4b8a8eb692f6c